envi/archs/h8/emu.py

```diff
@@ -0,0 +1,1314 @@
+        '''
+        va = emuProcessInterrupt_2140(intval)
+        self.setProgramCounter(va)
+
+    def emuProcessInterrupt_2140(self, intval=0):
+        '''
+        interrupt processing done according to H8/300 standards.  subclass and modify this function 
+        if the desired behavior differs.
+        supports: advanced and normal modes
+        '''
+        if self.getFlag(CCR_I):
+            # FIXME: check for NMI or RESET
+            if intval != 0 and intval not in NMI_INTS:
+                print("Ignoring maskable interrupt (CCR_I flag is set)")
+                return
+
+        print("Interrupt Handler: 0x%x" % intval)
+
+        addr = 0
+
+        if self.advanced:
+            if intval != 0:
+                val = (self.getStatusRegister() << 24) | self.getStatusRegister()
+                self.doPush(val, 4)
+
+                addr = (intval * 4) + 0xc
+
+            isrAddr = self.readMemortPtr(addr)
+        else:
+            if intval != 0:
+                val = self.getStatusRegister()
+                val |= (val << 8)
+                self.doPush(val, 2)
+                self.doPush(self.getProgramCounter())
+
+                addr = (intval * 2) + 6
+            
+            isrAddr = self.readMemoryPtr(addr)
+
+        return isrAddr
+
+    def emuProcessInterrupt(self, intval=0):
+        '''
+        interrupt processing done according to H8/300 standards.  subclass and modify this function 
+        if the desired behavior differs.
+        supports: advanced and normal modes
+        '''
+        print("Interrupt Handler: 0x%x" % intval)
+
+        addr = 0
+
+        if self.advanced:
+            if intval != 0:
+                self.doPush(self.getStatusRegister())
+                self.doPush(self.getStatusRegister())
+                self.doPush(self.getProgramCounter())
+
+                addr = (intval * 4) + 0xa
+
+            isrAddr = self.readMemortPtr(addr)
+        else:
+            if intval != 0:
+                self.doPush(self.getStatusRegister())
+                self.doPush(self.getProgramCounter())
+
+                addr = (intval * 2) + 4
+            
+            isrAddr = self.readMemoryPtr(addr)
+
+        return isrAddr
+        
+
+    def undefFlags(self):
+        """
+        Used in PDE.
+        A flag setting operation has resulted in un-defined value.  Set
+        the flags to un-defined as well.
+        """
+        self.setRegister(REG_EFLAGS, None)
+
+    def setFlag(self, which, state):
+        flags = self.getRegister(REG_FLAGS)
+        if state:
+            flags |= which
+        else:
+            flags &= ~which
+        self.setRegister(REG_FLAGS, flags)
+
+    def getFlag(self, which):
+        flags = self.getRegister(REG_FLAGS)
+        if flags == None:
+            raise envi.PDEUndefinedFlag(self)
+        return bool(flags & which)
+
+    def readMemValue(self, addr, size):
```

Comment:
Or maybe these helpers are only in one arch?  I forget...  if so, we should move them up to emu base class.